### ISSUE TITLE
[request] Export Highlights to a webDav folder

### ISSUE THREAD

**Issue by tobbsn** [koreader/koreader#5879]
I don't have an Evernote Account and I dont want to create one. So it would be very useful to export my highlights and annotations to somewhere else. Right now I have to exit koreader, activate USB is KSM, plug the device into my laptop, navigate to .adds/koreader/clipboard and manually copy all needed files somewhere. I am thinking of writing a small bashscript for KSM to use webdav somehow, but that doesn't seem practical either.

My kobo is already connected to my nextcloud via WebDAV and I think this could be a relatively easily implemented solution, since the protocol works in both directions.

**Comment by pazos** [comment]
If the export involves a computer we're already covered by @noembryo's [KoHighlights](https://github.com/noembryo/KoHighlights)

**Comment by tobbsn** [comment]
huh, I'll take a look at that. But my point was that I would like to skip the whole connecting a cable to my device and a computer part and use wifi sync instead

**Comment by pazos** [comment]
Actually a plugin that copies highlights for a given file and writes them to a markdown/html file would be cool, but that doesn't solve your issue, as far as I understand.

**Comment by tobbsn** [comment]
yes, in essence I am looking for a way to write such files to a webDav folder

**Comment by pazos** [comment]
I'm not an user of this plugin. I just skimed through the code and found it already supports exporting to:

local: txt, html
remote: evernote, some chinese app (evernote clone?) and joplin

I'll modify the tittle of the issue to match your demand +1

**Comment by pazos** [comment]
After #8944 is going to be way easier to write new export targets (including webdav).

Patches welcome for all of them.
But there's no plan to work specifically on webdav.

**Comment by fairlygood** [comment]
Four years later, admittedly, but I have made a webdav exporter for highlights:

https://github.com/fairlygood/koreader-highlights-webdav

**Comment by pazos** [comment]
@fairlygood: that's cool.

If you want to make things easier for your users you can take a look at https://github.com/koreader/contrib/tree/main/provider-old-exporters.koplugin

It showcases how to turn your `webdav` exporter into a thirdparty provider that you can distribute as a standalone plugin. That way users won't have to edit files after each update.
